lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,31 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception = @exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = []
+
+          while exception.respond_to?(:cause) && exception.cause
+            exception = exception.cause
+            cause = []
+            cause << '------------------'
+            cause << '--- Caused by: ---'
+            cause << "#{exception_class_name(exception)}:" unless exception_class_name(exception) =~ /RSpec/
+
+            encoded_string(exception.message.to_s).split("\n").each do |line|
+              cause << "  #{line}"
+            end
+
+            cause << ("  " + backtrace_formatter.format_backtrace(exception.backtrace, example.metadata).first)
+
+            potential_lines << cause
+          end
+
+          unless potential_lines.empty?
+            lines += potential_lines.last
+          end
```

Comment:
Good point, I'll update the pr to do the string generation for the last cause only :+1: